File: src/main/java/com/m14n/billib/data/BBDataParser.java
package com.m14n.billib.data;

import com.m14n.billib.data.model.BBTrack;
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;
import com.google.gson.stream.JsonReader;

import java.io.File;
import java.io.FileReader;
import java.io.IOException;

public class BBDataParser {
    public static void main(String... args) {
        final File theFolder = new File(BB.DATA_ROOT, "hot-100");
        final File[] files = theFolder.listFiles();
        final File theFile = files[files.length - 1];
        JsonReader theReader = null;
        try {
            theReader = new JsonReader(new FileReader(theFile));
            final GsonBuilder theBuilder = new GsonBuilder();
            final Gson theGson = theBuilder.create();
            theReader.beginArray();
            while (theReader.hasNext()) {
                BBTrack theTrack = theGson.fromJson(theReader, BBTrack.class);
                System.out.println(theTrack.toString());
            }
            theReader.endArray();
        } catch (IOException e) {
            System.err.println("Could not parse a file" + theFile.getName());
            e.printStackTrace();
        } finally {
            if (theReader != null) {
                try {
                    theReader.close();
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
        }

    }
}

File: src/main/java/com/m14n/billib/data/BBDataParser.kt
package com.m14n.billib.data

import com.m14n.billib.data.model.BBChart
import kotlinx.serialization.ImplicitReflectionSerializer
import kotlinx.serialization.UnstableDefault
import kotlinx.serialization.json.Json
import kotlinx.serialization.json.JsonConfiguration
import kotlinx.serialization.stringify
import java.io.File

@UseExperimental(UnstableDefault::class, ImplicitReflectionSerializer::class)
fun main() {
    JsonConfiguration.Stable.copy(prettyPrint = true)
    val chart = Json.parse(BBChart.serializer(), File(BB.DATA_ROOT, "hot-100").listFiles().last().readText())
    print(Json.stringify(chart))
    chart.tracks?.forEach { println(it.toString()) }
}
